mqtt/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@
 option(DOCUMENTATION "Build documentation?" OFF)
 
 option(WITH_STATIC_LIBRARIES "Build static versions of the libmosquitto/pp libraries?" OFF)
+option(WITH_PIC "Build the static library with PIC (Position Independent Code) enabled archives?" OFF)
```

Comment:
What for do you add this option? There is nothing about `WITH_PIC` in commit message.

Reply:
This is related to [CMP0077](https://cmake.org/cmake/help/latest/policy/CMP0077.html). I've added description of this in the commit message.

Reply:
If we'll not define it as an option explicitly, the effect of `set(WITH_PIC ON)` (several lines below, under `if (STATIC_BUILD)`) will be discarded by `option(WITH_PIC <...> OFF)` in `third_party/mosquitto/lib/CMakeLists.txt`.

Reply:
As for me, the reason in the commit message is wrong. Without this change `tarantoolctl rocks make` does not work for me:

```
[100%] Linking C shared library driver.so
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../third_party/mosquitto/build/lib/libmosquitto.a(net_mosq.c.o): warning: relocation against `_ui_method' in read-only section `.text'
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../third_party/mosquitto/build/lib/libmosquitto.a(net_mosq.c.o): relocation R_X86_64_PC32 against symbol `tls_ex_index_mosq' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

Now it is described like 'disable -fPIC by default', but:

1. The commit is about building from the rockspec and in context of `tarantoolctl rocks make` it is enabled. That's confusing.
2. It is already disabled by default when building with cmake directly.

Reply:
Fixed.

Reply:
> That is needed for CMake could not set this option off in CMake file of third
> party library while building the module statically when this option is enabled.

I doubt I read it correctly. Can you rephrase it in the form like the following:

> Without this change the behaviour is the following. If \<precondition\>, then \<bad result\>.

So it'll be easy to spot why it is needed.

Reply:
Fixed, now it look better.

Reply:
Looks okay for me now.

Reply:
(Will resolve this discussion.)